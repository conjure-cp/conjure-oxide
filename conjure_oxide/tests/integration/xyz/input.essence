find a,b,c : int(1..3)
such that a + b + c = 4
such that a >= b
