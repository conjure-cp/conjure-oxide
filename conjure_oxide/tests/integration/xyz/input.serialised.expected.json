{
  "variables": [
    [
      {
        "UserName": "b"
      },
      {
        "domain": {
          "IntDomain": [
            {
              "Bounded": [
                1,
                3
              ]
            }
          ]
        }
      }
    ],
    [
      {
        "UserName": "a"
      },
      {
        "domain": {
          "IntDomain": [
            {
              "Bounded": [
                1,
                3
              ]
            }
          ]
        }
      }
    ],
    [
      {
        "UserName": "c"
      },
      {
        "domain": {
          "IntDomain": [
            {
              "Bounded": [
                1,
                3
              ]
            }
          ]
        }
      }
    ]
  ],
  "constraints": [
    {
      "Eq": [
        {
          "Sum": [
            {
              "Reference": {
                "UserName": "a"
              }
            },
            {
              "Reference": {
                "UserName": "b"
              }
            },
            {
              "Reference": {
                "UserName": "c"
              }
            }
          ]
        },
        {
          "ConstantInt": 4
        }
      ]
    },
    {
      "Geq": [
        {
          "Reference": {
            "UserName": "a"
          }
        },
        {
          "Reference": {
            "UserName": "b"
          }
        }
      ]
    }
  ]
}
